Core/Src/main.c

```diff
@@ -143,7 +156,17 @@ void SystemClock_Config(void)
 }
 
 /* USER CODE BEGIN 4 */
+uint16_t read_adc() {
+	uint8_t adc_tx_data[3] = {0x01, 0x80, 0x00}; // transmission data based on ADC data sheet
```

Comment:
So although these are techincally still magic numbers, they are viewed as acceptable because we understand that they represent bits in a buffer, BUT for maximum readability you could also give these numbers a home, up to you.